lib/network/level-net/utils.js

```diff
@@ -0,0 +1,50 @@
+'use strict'; // eslint-disable-line
+
+/**
+ * @brief turn all @a err own and prototype attributes into own attributes
+ *
+ * This is done so that JSON.stringify() can properly serialize those
+ * attributes (e.g. err.notFound)
+ *
+ * @param {Error} err error object
+ * @return {Object} flattened object containing @a err attributes
+ */
+module.exports.flattenError = function flattenError(err) {
+    if (!err) {
+        return err;
+    }
+    const flattenedErr = {};
+
+    flattenedErr.message = err.message;
+    for (const k in err) {
+        if (!(k in flattenedErr)) {
+            flattenedErr[k] = err[k];
+        }
+    }
+    return flattenedErr;
+};
+
+/**
+ * @brief recreate a proper Error object from its flattened
+ * representation created with flattenError().
+ *
+ * @note Its internals may differ from the original Error object but
+ * its attributes should be the same.
+ *
+ * @param {Object} err flattened error object
+ * @return {Error} a reconstructed Error object inheriting @a err
+ *   attributes
+ */
+module.exports.reconstructError = function reconstructError(err) {
+    if (!err) {
+        return err;
+    }
+    const reconstructedErr = new Error(err.message);
+
+    for (const k in err) {
```

Comment:
This is super minor but I think it's good to stick to `Object.keys(obj).forEach` pattern of looping through object properties as this pattern performs slow when the object is in hashtable mode(lots property modifications, deletions etc).